src/components/Disputes/DisputeActions.js

```diff
@@ -406,4 +376,72 @@ const VoteInfo = ({ commitmentDate, outcome, revealDate }) => {
   )
 }
 
+// Assumes juror revealed vote
+function getAttributesWhenRevealed(
+  lastRound,
+  jurorDraft,
+  finalRulingConfirmed,
+  backgroundColor
+) {
+  const { appeal, vote } = lastRound
+
+  // Check if has voted in consensus with the plurality for the last round
+  const hasVotedInConsensus = vote && jurorDraft.outcome === vote.winningOutcome
+  // We must check if the penalties were already settled so we can tell the jurors
+  // wether their HNY locked balance has been discounted or they can claim rewards
+  // Note that if the penalties for the round are settled it means that the dispute has already ended
+  const settledPenalties = lastRound.settledPenalties
+
+  let background, icon, paragraph, title
+
+  // Juror voted in consensus during voting phase
+  if (hasVotedInConsensus) {
+    background = backgroundColor[appeal ? 'negative' : 'positive']
+    icon = appeal ? IconVotingFailed : IconRewardsGreen
+    paragraph = appeal ? (
```

Comment:
maybe we can have this as an use memo or maybe get the data from a function instead of doing so many `? :`

Reply:
Yeah i'll try to do an early return and see how it looks.
What do you mean getting the dats from a function? As you may notice i already separated this part in its own function, but there are so many different cases and conditions that i couldn't find a way to separate it and make it cleaner. 

But we can try to do the early return. Also regarding the useMemo, note that the function is already inside an useMemo

Reply:
created an issue #90